pkg/tfbridge/validate_input_types_test.go

```diff
@@ -1639,3 +1639,95 @@ func TestValidateInputType_toplevel(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateConfigType(t *testing.T) {
+	testCases := []struct {
+		name      string
+		inputName string
+		input     resource.PropertyValue
+		failures  []TypeFailure
+	}{
+		{
+			name:      "unexpected_argument",
+			inputName: "endpoints",
+			failures: []TypeFailure{{
+				Reason:       "an unexpected argument \"wxyz\" was provided",
+				ResourcePath: "endpoints[0]",
+			}},
+			input: resource.NewArrayProperty([]resource.PropertyValue{
+				resource.NewObjectProperty(resource.NewPropertyMapFromMap(map[string]interface{}{
+					"wxyz": "foo",
+				})),
+			}),
+		},
+		{
+			name:      "no_error_for_extra_inputs",
+			inputName: "doesnt_exist",
+			input: resource.NewArrayProperty([]resource.PropertyValue{
+				resource.NewObjectProperty(resource.NewPropertyMapFromMap(map[string]interface{}{
+					"wxyz": "foo",
+				})),
+			}),
+		},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			pspec := pschema.PackageSpec{
+				Name: "test",
+				Types: map[string]pschema.ComplexTypeSpec{
+					"aws:config/endpoints:endpoints": {
+						ObjectTypeSpec: pschema.ObjectTypeSpec{
+							Type: "object",
+							Properties: map[string]pschema.PropertySpec{
+								"abcd": {
+									TypeSpec: pschema.TypeSpec{
+										Type: "string",
+									},
+								},
+							},
+						},
+					},
+				},
+				Config: pschema.ConfigSpec{
+					Variables: map[string]pschema.PropertySpec{
+						"endpoints": {
+							TypeSpec: pschema.TypeSpec{
+								Type: "array",
+								Items: &pschema.TypeSpec{
+									Ref: "#/types/aws:config/endpoints:endpoints",
+								},
+							},
+						},
+					},
+				},
+			}
+			urn := resource.CreateURN(
+				"testResource",
+				"pkg:mod:ResA",
+				"",
+				"stack",
+				"project",
+			)
+
+			v := &PulumiInputValidator{
+				urn:                  urn,
+				schema:               pspec,
+				validateUnknownTypes: true,
+			}
+			failures := v.ValidateConfig(resource.PropertyMap{
+				resource.PropertyKey(tc.inputName): tc.input,
+			})
+			if failures != nil && len(failures) != len(tc.failures) {
+				t.Fatalf("%d failures, got %d: %v", len(tc.failures), len(failures), failures)
+			}
+			if len(tc.failures) > 0 {
+				if failures == nil {
+					t.Fatalf("expected failures, got none")
+				} else {
+					assert.Equal(t, tc.failures, failures)
+				}
+			}
```

Comment:
Why do we have so much validation logic on failures to assert that `tc.failures == failures`? Can't we just use `assert.Equal` to handle all of that?
```suggestion
			assert.Equal(t, tc.failures, failures)
```